canarytokens/aws_infra/operations.py

```diff
@@ -176,49 +190,55 @@ async def _build_handle_response_payload(
     response_content = (
         json.loads(handle.response_content) if handle.response_content else {}
     )
+    error = (
+        AWSInfraServiceError.parse(response_content.get("error", ""))
+        if not timeout
+        else AWSInfraServiceError.REQ_HANDLE_TIMEOUT
+    )
     payload = {
-        "result": not timeout and response_content.get("error", "") == "",
+        "result": error == AWSInfraServiceError.NO_ERROR,
         "handle": handle_id,
+        "message": service_error_map.get(error, "An unknown error occurred."),
```

Comment:
We should have an entry in the map for each error in the enum. Since `error` is definitely going to be of type `AWSInfraServiceError` we should cover each case explicitly in one place.
```suggestion
        "message": SERVICE_ERROR_MESSAGE_MAP[error],
```